github-extension/src/node/github-service-impl.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable, inject } from "inversify";
+import { GithubService, SshKeyPair, SshKeyServer } from '../common/github-service';
+import { Repository, Credentials, User, PullRequest, Organization, Collaborator } from '../common/github-model';
+
+var GitHubApi = require('github');
+
+@injectable()
+export class GithubServiceImpl implements GithubService {
+
+    constructor(@inject(SshKeyServer) protected readonly sshKeyServer: SshKeyServer) { }
+
+    getRepository(credentials: Credentials, owner: string, repository: string): Promise<Repository> {
+        return this.getConnection(credentials).repos.get({ owner: owner, repo: repository }).then((result: any) => {
+            return new Promise<Repository>(resolve => { resolve(result.data) })
+        });
+    }
```

Comment:
Why not use async/await syntax for all methods in GithubServiceImpl?
```typescript
async getRepository(credentials: Credentials, owner: string, repository: string): Promise<Repository> {
    const con = this.getConnection(credentials);
    const repo = await con.repos.get({ owner: owner, repo: repository });
    return repo.data;
}
```
The same three lines but much easier to read. WDYT?

Reply:
done